cmd/cmd.go

```diff
@@ -357,6 +360,47 @@ func RunHandler(cmd *cobra.Command, args []string) error {
 	return generateInteractive(cmd, opts)
 }
 
+func errFromUnknownKey(unknownKeyErr error) error {
+	// find SSH public key in the error message
+	sshKeyPattern := `ssh-\w+ [^\s"]+`
```

Comment:
I think this deserves a structured response. Parsing important details like this from a string is brittle. Something like this:

```json
{
  "error": "unauthorized: ...",
  "error_details": {
    "reason": "unknown key",
    "fingerprint": "..."
  }
}
```

Reply:
I agree, but none of our other errors are structured yet, so I would say its out of scope for this change.

Reply:
I think we can put a TODO in here and replace it in the future. We're the only people that would potentially break the string. Maybe put a comment in `server/images.go` as well? Definitely agree we need better error responses.